packages/powersync_core/lib/src/database/powersync_db_mixin.dart

```diff
@@ -110,10 +114,36 @@ mixin PowerSyncDatabaseMixin implements SqliteConnection {
     statusStream = statusStreamController.stream;
     updates = powerSyncUpdateNotifications(database.updates);
 
+    _manualSchemaManagementCompleted = false;
```

Comment:
I'm not super happy with this design, but I don't see a better option either.

One thing that looks cleaner might be to have users open their own databases (from `sqlite_async`), initialize the schema there and then pass the database to `PowerSyncDatabase.withDatabase`. But that would make the database tricky to use with wrappers like drift because we can't swap out the earlier database with the other one later.
So this might be the best option we have here, but I'll ping @rkistner for API design advice as well.